sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaUnboundedReader.java

```diff
@@ -77,6 +77,8 @@
  */
 class KafkaUnboundedReader<K, V> extends UnboundedReader<KafkaRecord<K, V>> {
 
+  boolean atleastOnePollCompleted = false;
```

Comment:
will this ever be access by multiple threads? or is the kafka reader single threaded?

might be a good idea to mark the class as `@NotThreadSafe`

Reply:
Done